src/Stratis.Bitcoin.Features.Consensus/Rules/ProvenHeaderRules/ProvenHeaderRuleBase.cs

```diff
@@ -23,10 +24,8 @@ public override void Initialize()
 
             Guard.NotNull(this.PosParent, nameof(this.PosParent));
 
-            if (this.PosParent.Network.Consensus.Options is PosConsensusOptions options)
-            {
-                this.ProvenHeadersActivationHeight = options.ProvenHeadersActivationHeight;
-            }
+            var checkpoints = new Checkpoints(this.PosParent.Network, this.PosParent.ConsensusSettings);
```

Comment:
We have checkpoints on the parent `this.PosParent.Checkpoints`

Reply:
should be the same list right?
btw I think we should use ICheckpoints everywhere, too many ways to access checkpoints and you could change it too accessing directly the dictionary, what do you think?
read this issue I opened few days ago: https://github.com/stratisproject/StratisBitcoinFullNode/issues/2663